plugins/package-managers/cargo/src/main/kotlin/Cargo.kt

```diff
@@ -191,14 +192,19 @@ class Cargo(
         )
 
         val nonProjectPackages = packageById.values.mapNotNullTo(mutableSetOf()) { cargoPkg ->
-            cargoPkg.takeUnless { it.isProject() }?.toPackage(hashes)
+            cargoPkg.takeUnless { it.isProject(workingDir) }?.toPackage(hashes)
         }
 
         return listOf(ProjectAnalyzerResult(project, nonProjectPackages))
     }
 }
 
-private fun CargoMetadata.Package.isProject() = source == null
+private fun CargoMetadata.Package.getLocalPath(): File? =
+    id.substringAfter("path+file://", "").ifEmpty { null }
+        ?.removeSuffix(")")?.substringBefore("#")?.let { File(it) }
+
+private fun CargoMetadata.Package.isProject(root: File): Boolean =
+    source == null && getLocalPath()?.startsWith(root) == true
```

Comment:
Can we have a code comment, explaining the idea besides the expression ?

Reply:
I've extracted a speaking variable instead.